subsys/net/ip/ipv4.c

```diff
@@ -172,6 +173,7 @@ enum net_verdict net_ipv4_process_pkt(struct net_pkt *pkt)
 	}
 
 drop:
+	NET_DBG("pkt len %d dropped", real_len);
```

Comment:
in-case of mismatch lengths, it shows two debug logs for same purpose.
DROP: pkt len per hdr %d != pkt real len %d
pkt len %d dropped

Are two debug logs necessary for same purpose?

Reply:
Well, "necessary" is not the best word here. It's just as I explained in my initial comments to https://github.com/zephyrproject-rtos/zephyr/pull/11407, I don't really call for *every* drop reason to be logged, just *most realistic* ones. For the rest, can start with a "catch all" logging like @tbursztyka suggested here: https://github.com/zephyrproject-rtos/zephyr/pull/11407#issuecomment-440254727 . 

So yes, that would mean that in some cases we'd have 2 logging entries per drop. I think that's ok for now. We can make further elaborations to it as we go, e.g. could have:

~~~
drop_logged:
  NET_DBG(...);
drop_silent:
  return NET_DROP;
~~~

Reply:
Actually, I kind of changed my mind about logging drop reasons: It's better to be specific rather than generic.

So heare, instead, let's have proper NET_DBG message right before each goto drop, and not one at label drop.

Reply:
Well, being too detailed also has its issue - we'll soon run out of flash for debug builds ;-). And it's hard to be completely specific in this patch, because some verdicts come from next layers of the stack. Anyway, let me look how to improve it based on this discussion.

Reply:
> Anyway, let me look how to improve it based on this discussion.

Oh, indeed need https://github.com/zephyrproject-rtos/zephyr/pull/11407 to be merged first, as it changes some cases in this func already.

Reply:
I guess the one which generates the drop verdict, should be the only one having the dbg message. And maybe for ENOMEM stuff, it's useless to log (but in net_pkt/frag allocator then).
I forgot about ROM issue, that might be a problem indeed.